src/workers/dominantColor.worker.js

```diff
@@ -83,6 +86,48 @@ export default () => {
       })
   }
 
+  const findDominantColors = selectFrom => {
+    const domColors = [selectFrom.shift()]
+
+    while (domColors.length < NUM_DOMINANT_COLORS && selectFrom.length > 0) {
+      const indexOfNextDomColor = findIndexOfMaxEuclideanDistance(
+        domColors,
+        selectFrom
+      )
+      domColors.push(selectFrom[indexOfNextDomColor])
+      selectFrom.splice(indexOfNextDomColor, 1)
+    }
+    return domColors
+  }
+
+  const findIndexOfMaxEuclideanDistance = (existing, selectFrom) => {
```

Comment:
the leetcode goes to show. this is snazzy.
I don't think there's a huge perf gain for it, but you *could* probably memoize some of the euclidean distances along the way to save future computation if you see the same pair again

biggest advantage here is that this runs in a web worker so it has an isolated js context and wont block rendering on the main thread